### PR TITLE
libSceHttp early implementation

### PR DESCRIPTION
**Pull request by minecraft8997**
Not really tested, but it should support GET and POST requests at basic level.

### CODE REVIEW

File: modules/libSceHttp/entry.cpp [hunk 1]
```diff
@@ -5,74 +5,439 @@
...
+#include <boost/asio.hpp>
+
+using namespace boost::asio;
+
 LOG_DEFINE_MODULE(libSceHttp);
 
 namespace {
 using SceHttpsCallback = SYSV_ABI int (*)(int libsslCtxId, unsigned int verifyErr, SceSslCert* const sslCert[], int certNum, void* userArg);
 
+#define ARRAY_LENGTH 128
```
Comment (SysRay) [inline modules/libSceHttp/entry.cpp:20]: constexpr func = [](auto container, int id){} ? use std::array instead and call size()
or 
template<typename T, int size>
int testFunc(T(&)[size] container, int id){}

File: modules/libSceHttp/entry.cpp [hunk 2]
```diff
@@ -5,74 +5,439 @@
...
+
+#define TEST_HTTP_CTX_ID TEST_ID(g_clients, libhttpCtxId)
+
+#define TEST_TMPL_ID TEST_ID(g_templates, tmplId)
+
+#define TEST_CONN_ID TEST_ID(g_connections, connId)
+
+#define TEST_REQ_ID TEST_ID(g_requests, reqId)
+
+#define LOOKUP_AVAILABLE_IDX(ARRAY, ACTION)                                                                                                                    \
```
Comment (SysRay) [inline modules/libSceHttp/entry.cpp:32]: same here

**Comment by minecraft8997** [comment]
Added new commits.

**Comment by minecraft8997** [comment]
Found a small mistake about calling testId(), will fix it.

**Comment by SysRay** [comment]
Try to use one class instead and let it manage all the request/template etc resources, the dependencies are just to much.

maybe use std::unique_ptr<> or std::shared_ptr (requests) instead of new/delete.

**Pull request merged**